code_docs/Organization.md

```diff
@@ -44,6 +44,21 @@ The documentation structure would look something like this:
 
 
 ### Code Cleanup
+#### Automation View Refactoring
+**Proposal:**
+When refactoring it would make sense to create a class `AutomationEditor` which just implements the editor itself. Ideally this class would implement a few different layouts for the editing interface, for example the velocity editing layout. When launching the editor, the layout can be set appropriately to the parameter edited.
+
+If necessary, subclasses could be created to implement different behavior. But as all the editors look extremely similar, I feel like having them all in one class would be most efficient.
+
+A separate class `AutomationEditorLaunchpad` would implement the launchpad.
+
+`AutomationEditorLaunchpad` presents different sets of parameters to edit depending on where/how it is launched. Instrument Clip View => [audition pad] + [song] could show Note Automation.
+
+Considerations:
+- Note automation like probability could be automated on clip *and* note level. (Which parameters should be automatable and how is a different discussion.)
```

Comment:
Do you mean like edit the probability of all notes vs a single note? yea that's been discussed as well.

There's several scenarios here:
- edit the probability of a step (all notes in that step)
- edit the probability of a step (all notes in that step) across all rows (e.g. pitches / drums) in a clip
- edit the probability of all notes in a row
- edit the probability of all notes in a clip

Reply:
Exactly. I was mainly pointing out that having different levels of launchpad (clip level and note level) creates a clear place to store both levels of automation. This differs from the current implementation where velocity automation of note rows is mapped to a pad in the launchpad on clip level

Reply:
> Exactly. I was mainly pointing out that having different levels of launchpad (clip level and note level) creates a clear place to store both levels of automation. This differs from the current implementation where velocity automation of note rows is mapped to a pad in the launchpad on clip level

Ah that's a good point about the launchpad mixing the row level velocity editor shortcut with the clip level parameter shortcut.

This happened mostly because we have not implemented the concept of turning Affect Entire on / off like Kit clips. The idea was to have two layers (affect entire - clip level, and not affect entire - row level)